Implement OEmbed provider for JSON

tcmorris opened the pull request:
Implement OEmbed provider for JSON and add extra options for embedded media
e.g. Instagram, IFTTT, Rdio, Kickstarter

Related issue: http://issues.umbraco.org/issue/U4-6562

Shazwazza commented on src/Umbraco.Core/Media/IEmbedProvider.cs:
```diff
@@ -2,7 +2,7 @@
     public interface IEmbedProvider
     {
-        bool SupportsDimensions { get; }
+        bool SupportsDimensions { get; set; }
```
This interface shouldn't change, there's no reason to make this setter public and part of the interface

Shazwazza commented on src/Umbraco.Web/Media/EmbedProviders/AbstractOEmbedProvider.cs:
```diff
@@ -20,7 +25,7 @@ public virtual bool SupportsDimensions
         public abstract string GetMarkup(string url, int maxWidth, int maxHeight);
 
-        public virtual string BuildFullUrl(string url, int maxWidth, int maxHeight)
+        public virtual string BuildRequestUrl(string url, int maxWidth, int maxHeight)
```
Unfortunately things like this are breaking API changes which we cannot do, are you able to adjust this code so there are no breaking changes?

Shazwazza commented:
Thanks for the PR. Before we can pull this in there is a few things I'm hoping is possible:

* Create an issue on tracker with details about the feature request and a link to this PR so we can track it
* Ensure there are no breaking changes - there would be various ways to do this but we always must take this into account so that peoples sites and packages are not broken

tcmorris commented:
Updated the code for the comments you mentioned. Also added mappings for JSON properties which I forgot about before. The link to the issue on tracker is also above. Added a link for the PR on the issue itself, let me know if all okay.

Shazwazza commented:
Cool, i'll tag it for a 7.3 release and hope it makes it in there, will review when i get a chance. Thanks!!

Shazwazza commented:
We've finally pulled this all in  :)

I made a change in rev: fdf53b85501d43923733845ec503e7114fc4955b

because removing the flickr provider is a breaking change.

Shazwazza commented on src/Umbraco.Web/Media/EmbedProviders/OEmbedVideo.cs:
```diff
@@ -6,12 +6,10 @@ public class OEmbedVideo : AbstractOEmbedProvider
+            string requestUrl = BuildFullUrl(url, maxWidth, maxHeight);
+
+            XmlDocument doc = GetXmlResponse(requestUrl);
+            return GetXmlProperty(doc, "/oembed/Html");
```
By the way, this case change breaks the providers. needs to be lower case Html!

tcmorris replied:
Hmm not sure why I changed that, good spot!

Shazwazza replied:
Have checked the rest, all works great!

tcmorris commented:
Great thanks for sorting that out Shannon :)
